src/SingleSignOnDemo/SSOExample.cs

```diff
@@ -0,0 +1,129 @@
+        /// Executer le procès d'authentification avec le jeton donné.
+        /// </summary>
+        /// <param name="token">Das Token, das für den Login verwendet werden soll. The token used to execute the login process. Le signe utilisé pour executer le procès d'authentification.</param>
+        /// <returns></returns>
+        public static bool Login(string token)
+        {
+            try
+            {
+                // Eine Nonce erstellen, um ein Box-Objekt erstellen zu können.
+                // Create a nonce used to create a Box object.
+                // Créer un nonce utilisé pour créer un objet Box.
+                byte[] nonce = PublicKeyBox.GenerateNonce();
+
+                // Einen 64 Byte großen Puffer erstellen und ihn mit zufälligen Bytes füllen.
+                // Create a 64-byte buffer array and fill it with random bytes.
+                // Créer un tableau tampon de 64 octets et remplisser-le avec des octets aléatoires.
+                byte[] buffer = new byte[64];
+                new Random().NextBytes(buffer);
+
+                // Ein Box-Object erzeugen, die zur Kommunikation mit mySMS unter Verwendung des öffentlichen und privaten Schlüssels dient.
+                // Create a Box object to communicate with mySMS using the public and private key.
+                // Créer un object Box pour communiquer avec mySMS en utilisant les clés publique et privée.
+                byte[] encrypted = PublicKeyBox.Create(buffer, nonce, Convert.FromBase64String(APPLICATION_PRIVATE_KEY), Convert.FromBase64String(PLATFORM_PUBLIC_KEY));
+
+                // Nonce und verschlüsselten Text aneinanderreihen und in einen Base64-String umwandeln.
+                // Concatenate the nonce and the encrypted text and convert it to a base64 string.
+                // Concaténer le nonce et le texte chiffré et convertisser-le en chaîne base64.
+                byte[] payload = nonce.ToList().Concat(encrypted.ToList()).ToArray();
+                string base64 = Convert.ToBase64String(payload, Base64FormattingOptions.None);
+
+                // Eine HTTP-Anfrage an die Plattform senden.
+                // Send an HTTP request to the platform.
+                // Envoyer une requête HTTP à la plate-forme.
+                string sso_url = string.Format("{0}/auth/lookup/{1}", SSO_BASE_URL, token);
+                HttpWebRequest req = WebRequest.CreateHttp(sso_url);
+                req.Headers.Add("Authorization", string.Format("PRODUCTAUTH {0}:{1}", PRODUCT_NAME, base64));
+                req.Method = "GET";
+
+                // Die Antwort empfangen und verarbeiten
+                // Receive and proceed the response.
+                // Recever et passer la réponse.
+                HttpWebResponse resp = (HttpWebResponse)req.GetResponse();
+                if (resp.StatusCode == HttpStatusCode.OK)
+                {
+                    using (MemoryStream ms = new MemoryStream())
```

Comment:
as you are using the MemoryStream only as a helper to turn the response in the array, you could call ToArray() immediately, and close the `using` block. This would save you an indentation level, as well as the duplicated ToArray()-call.